### PR TITLE
Added clearFilters method to allow re-use of same initialized image

### PR DESCRIPTION
**Pull request by virtuman**
Added clearFilters method to reset all currently applied filters to allow re-usage of init'ed image and object. This allows to create output for same image with different filters applied to it.

### CODE REVIEW

**Comment by virtuman** [comment]
I can't merge it in, will you be able to merge this change in ? thank you.

**Comment by harto** [comment]
Could you provide an example of what you need this for?

**Comment by virtuman** [comment]
a simple example would be to use zoomed images with responsive images similar to:
$img = Thumbor\Url\Builder::construct($server, $secret, 'http://images.example.com/llamas.jpg');

fitIn(400, 400)->build() ?>" src-set="<?= $img->fitIn(800, 800)->trim()->addFilter('no_upscale', null)->build() ?> x2, <?= $img->fitIn(1200, 1200)->trim()->build() ?> x3" data-zoom-image="<?= $img->fitIn(800, 800)->clearFilters()->build() ?>" />

so this would allow me to re-use the same initialized image without having to init it for every instance within  tag. all other options are overwrite-able, but filters currently can only be appended to the list with addFilter.

**Comment by pda** [comment]
Perhaps making `Builder` immutable would work better?
Calls like `->fitIn()` and `->addFilter()` could return a new Builder instance with that operation added.
That way builder operations only effect the resulting builder object, and don't leak backwards into the builder state.

```php
$img = Thumbor\Url\Builder::construct($server, $secret, 'http://example.com/llama.png');
$small = $img->fitIn(400, 400);
$zoom = $img->fitIn(800, 800);

$small->build(); // your 400x400
$zoom->trim()->addFilter('no_upscale', null)->build(); // 800x800 filtered
$zoom->build(); // 800x800 unfiltered
```

**Comment by virtuman** [comment]
That would work, a little too wordy and I was concerned mostly by the fact that I didn't want to affect any existing users of phumbor, so it felt like adding a clear() method is a safer bet for keeping less verbose syntax and to cover more grounds. 

Would merging my suggestion still be possible or should i switch over to use my own fork of the brunch at this point?

**Comment by harto** [comment]
I think making the builder return a copy of itself is better, although it would be a breaking change. If you submit a new PR with that implementation I'd release it as a new major version.

**Comment by virtuman** [comment]
so no go on this pull request?

**Comment by harto** [comment]
Not in this form, but I'd consider a patch in the other form discussed.
